websocket-subscription-2021.html

```diff
@@ -0,0 +1,499 @@
+                  <p><em>This section is non-normative.</em></p>
+
+                  <p>An example <code>POST</code> request using a <code>DPoP</code> bound access token is below:</p>
+
+                  <figure id="notification-subscription-request" class="example listing" rel="schema:hasPart" resource="#notification-subscription-request">
+                    <p class="example-h"><span>Example</span>: Requesting an WebSocket subscription URL from the Notification Subscription.</p>
+
+                    <pre about="#notification-subscription-request" property="schema:description" typeof="fabio:Script"><code>POST /subscription</code>
+<code>Authorization: DPoP &lt;token&gt;</code>
+<code>DPoP: &lt;proof&gt;</code>
+<code>Content-Type: application/ld+json</code>
+<code></code>
+<code>{</code>
+<code>  "@context": ["https://www.w3.org/ns/solid/notification/v1"],</code>
+<code>  "type": "WebSocketSubscription2021",</code>
+<code>  "topic": "https://storage.example/resource",</code>
+<code>  "state": "opaque-state",</code>
+<code>  "expiration": "2021-12-23T12:37:15Z",</code>
+<code>  "rate": "PT10s"</code>
+<code>}</code></pre>
+
+                    <figcaption property="schema:name"><code>POST</code> request including <code>type</code> and <code>topic</code> targeting the Notification Subscription API.</figcaption>
+                  </figure>
+
+                  <p>A successful response will contain a URL to the subscription API that can be used directly with a JavaScript client.</p>
+
+                  <figure id="notification-subscription-response" class="example listing" rel="schema:hasPart" resource="#notification-subscription-response">
+                    <p class="example-h"><span>Example</span>: Response from the Notification Subscription including the WebSocket subscription URL.</p>
+
+                    <pre about="#notification-subscription-response" property="schema:description" typeof="fabio:Script"><code>HTTP/2</code>
+<code>Content-Type: application/ld+json</code>
+<code></code>
+<code>{</code>
+<code>  "@context": "https://www.w3.org/ns/solid/notification/v1",</code>
+<code>  "type": "WebSocketSubscription2021",</code>
+<code>  "endpoint": "https://websocket.example/?auth=Ys3KiUq"</code>
+<code>}</code></pre>
+
+                    <figcaption property="schema:name">Response to the <code>POST</code> request including <code>type</code> and <code>endpoint</code>.</figcaption>
+                  </figure>
+
+                  <p>In JavaScript, a client can use the data in the response to establish a connection to the WebSocket hypermedia API.</p>
+
+                  <figure id="websocket-new" class="example listing" rel="schema:hasPart" resource="#websocket-new">
+                    <p class="example-h"><span>Example</span>: Establishing a new WebSocket connection.</p>
+
+                    <pre about="#websocket-new" property="schema:description" typeof="fabio:Script"><code>const ws = new WebSocket(url);</code></pre>
```

Comment:
Per [RFC 6455](https://datatracker.ietf.org/doc/html/rfc6455#section-4.1):

```
    6. If the response includes a |Sec-WebSocket-Protocol| header field
       and this header field indicates the use of a subprotocol that was
       not present in the client's handshake (the server has indicated a
       subprotocol not requested by the client), the client MUST _Fail
       the WebSocket Connection_.
```

Therefore, if a client asks for a certain subprotocol, the server cannot just pick some other subprotocol. A client may indicate support for multiple subprotocols: a, b, and c; in that case, the server can pick any of these.

Yes, it would be good to include some language specifically about this.